lib/api/apiUtils/object/abortMultipartUpload.js

```diff
@@ -105,9 +112,47 @@ function abortMultipartUpload(authInfo, bucketName, objectKey, uploadId, log,
                     }
                     cb();
                 });
+            }, () => next(null, mpuBucket, storedParts, destBucket, objectMD));
+        },
+        function deleteObjectMetadata(mpuBucket, storedParts, destBucket, objectMD, next) {
+            if (!objectMD || metadataValMPUparams.uploadId !== objectMD.uploadId) {
+                return next(null, mpuBucket, storedParts, destBucket, objectMD);
+            }
+            // In case there has been an error during cleanup after a complete MPU
+            // (e.g. failure to delete MPU MD in shadow bucket),
+            // we need to ensure that the MPU metadata is deleted.
+            log.info('Object has existing metadata, deleting them', {
+                method: 'abortMultipartUpload',
+                bucketName,
+                objectKey,
```

Comment:
It could be good to log the `versionId` and `uploadId` as well (but maybe it's already in the log context?)

Reply:
unfortunately, I am not sure it is so simple, to cover all situations (versioned, non-versioned and version-suspended) : it probably works for both versioned and non-versioned cases, but i think it may break for the version-suspended case...

Also not sure if the field needs to be decoded/encoded here...

@Kerkesni Can you confirm? There helper functions to cover all cases, right?